services/AttendanceService.ts

```diff
@@ -27,13 +27,22 @@ export default class AttendanceService {
     return attendances.map((attendance) => attendance.getPublicAttendance());
   }
 
-  public async getAttendancesForUser(user: UserModel): Promise<PublicAttendance[]> {
+  public async getAttendancesForCurrentUser(user: UserModel): Promise<PublicAttendance[]> {
     const attendances = await this.transactions.readOnly(async (txn) => Repositories
       .attendance(txn)
       .getAttendancesForUser(user));
     return attendances.map((attendance) => attendance.getPublicAttendance());
   }
 
+  public async getAttendancesForUser(uuid: Uuid): Promise<PublicAttendance[]> {
+    return this.transactions.readOnly(async (txn) => {
+      const user = await Repositories.user(txn).findByUuid(uuid);
```

Comment:
Make sure to validate user isn't null before proceeding to access the fields of user